### DISCUSSION TITLE
ConnectKitButton.Custom should not be using children props

### DISCUSSION THREAD

**Discussion by GabrielCartier** [family/connectkit#23] [Ideas]
It's generally not a good idea to use `children` props, it is recommended to use children component instead. There are some `eslint` rules that will actually warn you about that (`react/no-children-prop`).

I think passing the children directly into the `ConnectKitButton.Custom` would be a more react design.
It would mean you would have to pass some of the props from one component to another but you could have all those props autocomplete in an easy way vs. using the `renderProps`.

**Comment by tofuness** [comment]
@GabrielCartier Hey, thanks for raising!

Right now `ConnectKitButton.Custom` _doesn't_ require that you pass in a `children` prop and it follows this pattern:
```
<ConnectKitButton.Custom>
  {(props) => (
    <p>Your button</p>
  )}
</ConnectKitButton.Custom>
```

**Reply by GabrielCartier** [reply]
Ah ok, my bad then, that works fine I guess. You guys can close!
